src/region_growing.jl

```diff
@@ -0,0 +1,208 @@
+
+default_diff_fn{CT1<:Colorant, CT2<:Colorant}(c1::CT1,c2::CT2) = sqrt(sum(abs2,(c1)-Images.accum(CT2)(c2)))
+
+"""
+    seg_img = seeded_region_growing(img, seeds, [kernel_dim], [diff_fn])
+    seg_img = seeded_region_growing(img, seeds, [neighbourhood], [diff_fn])
+
+Segments the N-D image `img` using the seeded region growing algorithm
+and returns a [`SegmentedImage`](@ref) containing information about the segments.
+
+# Arguments:
+* `img`             :  N-D image to be segmented (arbitrary indices are allowed)
+* `seeds`           :  `Vector` containing seeds. Each seed is a Tuple of a
+                       CartesianIndex{N} and a label. See below note for more
+                       information on labels.
+* `kernel_dim`      :  (Optional) `Vector{Int}` having length N whose ith element
```

Comment:
Maybe also allow `NTuple{N, Int}`? For small lists tuples are almost always more efficient (not that it matters here).